app/bundles/LeadBundle/Entity/LeadDeviceRepository.php

```diff
@@ -47,48 +47,75 @@ public function getTableAlias()
 
     /**
      * @param      $lead
-     * @param null $deviceName
-     * @param null $deviceBrand
-     * @param null $deviceModel
+     * @param null $deviceNames
+     * @param null $deviceBrands
+     * @param null $deviceModels
      *
      * @return array
      */
-    public function getDevice($lead, $deviceName = null, $deviceBrand = null, $deviceModel = null)
+    public function getDevice($lead, $deviceNames = null, $deviceBrands = null, $deviceModels = null, $deviceOss = null)
     {
         $sq = $this->_em->getConnection()->createQueryBuilder();
         $sq->select('es.id as id, es.device as device, es.device_fingerprint')
             ->from(MAUTIC_TABLE_PREFIX.'lead_devices', 'es');
         if (!empty($statIds)) {
             $inIds = (!is_array($statIds)) ? [(int) $statIds] : $statIds;
 
-            $sq->where(
+            $sq->andWhere(
                 $sq->expr()->in('es.id', $inIds)
             );
         }
 
-        if ($deviceName !== null) {
-            $sq->where(
-                $sq->expr()->eq('es.device', ':device')
-            )
-                ->setParameter('device', $deviceName);
+        if ($deviceNames !== null) {
+            if (!is_array($deviceNames)) {
+                $deviceNames[] = $deviceNames;
```

Comment:
This will throw a fatal `Fatal error: Uncaught Error: [] operator not supported for strings`

Should be `$deviceNames = [$deviceNames];` or `$deviceNames = (array) $deviceNames;`